analyses/cell-type-ewings/scripts/utils/tumor-validation-helpers.R

```diff
@@ -199,3 +237,86 @@ plot_cnv_heatmap <- function(
   )
   return(heatmap)
 }
+
+# create heatmap with gene set as rows and cells as columns 
+# this includes adding an annotation column labeling cells 
+# colors are automatically determined using the `Dark2` palette and assigning to cell types listed in the `annotation_column`
+full_celltype_heatmap <- function(classification_df,
+                                  gene_exp_columns,
+                                  annotation_column){
+  
+  # get list of all cell types being plotted and assign colors from Dark2 palette 
+  cell_types <- unique(classification_df[[annotation_column]])
+  num_cell_types <- length(cell_types)
+  colors <- palette.colors(palette = "Dark2") |> 
+    head(n = num_cell_types) |> 
+    purrr::set_names(cell_types)
+  
+  # create annotation for heatmap
+  annotation <- ComplexHeatmap::columnAnnotation(
+    singler = classification_df[[annotation_column]],
+    col = list(
+      singler = colors
+    )
+  )
+  
+  # build matrix for heatmap cells x gene set sum or mean 
+  heatmap_mtx <- classification_df |>
+    dplyr::select(barcodes, gene_exp_columns) |> 
+    tibble::column_to_rownames("barcodes") |> 
+    as.matrix()
+  colnames(heatmap_mtx) <- stringr::str_remove(colnames(heatmap_mtx), "_sum|mean-")
+  
+  # plot heatmap of marker genes 
+  plot_gene_heatmap(t(heatmap_mtx), 
```

Comment:
I'd probably move this `t()` up to the processing pipeline above. (and then change `colnames()` to `rownames()`, I guess)